sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -950,9 +950,15 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog {
       case f: Filter =>
         val (correlated, _) = splitConjunctivePredicates(f.condition).partition(containsOuter)
 
-        // Find any non-equality correlated predicates
+        // Find any non-equality correlated predicates and equality predicates that do not
+        // guarantee one-on-one mapping between inner and outer attributes. E.G:
+        // a = outer(c) -> true
```

Comment:
The example "true" here means it is a 1:1 mapping or not?

Reply:
Seems below `true` means it is not, but the example here `true` means it is. Easier to be confusing.

Reply:
Agree, we should refine this.